examples/README.md

```diff
@@ -0,0 +1,169 @@
+# Examples for meltano-map-transformer
+
+Either clone this repository, or just open it up in codespaces.
+
+1. cd to this directory if you haven't
+2. Run ```meltano install```
+3. Feel free to run the pipeline without a mapper ```meltano run tap-csv target-sqlite```
+5. check your database with ```sqlite3 -markdown output/tap_csv.db 'select * from customers limit 20'```
+
+To clean up afterwards, just remove the file:  ```rm output/tap_csv.db```
+
+# Example 1: Hashing with built-in functions
+**What:** Splitting email adresses, save the domain in separate, hash the mail itself, remove "other" columns and use the "config" object.
+
+**The configuration:**
+```yaml
+  mappers:
+  - name: meltano-map-transformer
+    pip_url: ../.
+    executable: meltano-map-transform
+    settings:
+    - name: stream_maps
+      kind: object
+    mappings:
+    - name: hash_email
+      config:
+        stream_maps:
+          customers:
+            id: id
+            email:      # drop the PII field from RECORD and SCHEMA messages
+            email_domain: email.split('@')[-1]
+            email_hash: md5(config['hash_seed'] + email)
+            __else__: null
+        stream_map_config:
+          hash_seed: 01AWZh7A6DzGm6iJZZ2T
+```
+
+**Steps:**
+1. Run ```meltano run tap-csv hash_email target-sqlite```
+2. Look at the data ```sqlite3 -markdown output/tap_csv.db 'select * from customers limit 20'```
```

Comment:
@sbalnojan what do you think about adding a ~1 record example input and output to each of these examples? Its clear for someone who clones/installs/runs the examples but if they dont have the time or ability to run them it would be nice to see a preview, something with content like (formatting TBD):

```markdown

### Input
id,first_name,last_name,email,ip_address
1,Ethe,Book,ebook0@twitter.com,67.61.243.220
...

### Output
email_domain  email_hash                        id          __loaded_at               
------------  --------------------------------  ----------  --------------------------
twitter.com   a283215a8272692d77dec9138e996c04  1           2023-03-17 14:19:46.214774
...
```

Here are the samples in case its helpful, since I'm already going through and testing them:
```
### Ex 2
description  id          __loaded_at               
-----------  ----------  --------------------------
Book         1           2023-03-17 14:29:15.257726

### Ex 3
email               first_name  id          ip_address     last_name   mapping_time  __loaded_at               
------------------  ----------  ----------  -------------  ----------  ------------  --------------------------
ebook0@twitter.com  Ethe        1           67.61.243.220  Book        2023-03-17    2023-03-17 14:31:50.454788

### Ex 4
client_id   email               first_name  hash_id                           id          ip_address     last_name   __loaded_at               
----------  ------------------  ----------  --------------------------------  ----------  -------------  ----------  --------------------------
client_123  ebook0@twitter.com  Ethe        c4ca4238a0b923820dcc509a6f75849b  1           67.61.243.220  Book        2023-03-17 14:33:28.225243

### Ex 5
count_t     email               first_name  id          last_name   __loaded_at               
----------  ------------------  ----------  ----------  ----------  --------------------------
0           EBOOK0@TWITTER.COM  ethe        1           book        2023-03-17 14:35:01.229091

### Ex 6
first_name  id          __loaded_at               
----------  ----------  --------------------------
found_book  1           2023-03-17 14:35:32.223231
no_book     2           2023-03-17 14:35:32.223335

### Ex 7
first_name  id          last_name   __loaded_at               
----------  ----------  ----------  --------------------------
Ethe        1           mask-1      2023-03-17 14:35:56.096918
Myranda     2           mask-2      2023-03-17 14:35:56.097042
...
Serge       5           mask-3      2023-03-17 14:35:56.097279
....
Giffy       7           mask-4      2023-03-17 14:35:56.097433
Vitoria     8           no-mask-fo  2023-03-17 14:35:56.097508
...
Doreen      13          Mask-2      2023-03-17 14:35:56.097884
```

Reply:
Yes! I love it, that's a great idea. I'll add that @pnadolny13 .